docs/pages/eas-update/migrate-codepush-to-eas-update.mdx

```diff
@@ -0,0 +1,75 @@
+---
+title: Migrating from CodePush to EAS Update
+description: A guide to help migrate from CodePush to EAS Update.
+---
+
+import { Collapsible } from '~/ui/components/Collapsible';
+import { Terminal } from '~/ui/components/Snippet';
+import { Step } from '~/ui/components/Step';
+
+Transition a [bare](/bare/hello-world/) React Native project with CodePush to use EAS Update. This guide assumes that you are using the default React Native project structure.
+
+## Prerequisites
+
+<Collapsible summary="An Expo user account">
+
+EAS Update is available to anyone with an Expo account, regardless of whether you pay for EAS or use our free tier. You can sign up at [https://expo.dev/signup](https://expo.dev/signup).
+
+Learn more about different plans and benefits at [EAS pricing](https://expo.dev/pricing).
+
+</Collapsible>
+
+<Step label="1">
+## Uninstall CodePush
+
+To avoid conflicts and unexpected behavior, it's recommended to uninstall CodePush if you're using EAS Update. This is because your app could periodically fetch updates from both services, leading to issues, especially if you're using different configurations for each service.
+
+Remove the CodePush SDK from your project by uninstalling the `react-native-code-push` package:
+
+<Terminal cmd={['$ npm uninstall react-native-code-push']} />
+
+You'll also need to remove CodePush references from JS and native code. Follow this [guide](https://github.com/Microsoft/react-native-code-push/issues/1101#issuecomment-350204507) for more detailed instructions.
+
+</Step>
+
+<Step label="2">
+## Ensure `app.json` exists
+
+Ensure that your project has an `app.json` file. If you don't have one, you can create one by running the command:
+
+<Terminal
+  cmd={[
+    "# If app.json doesn't exist, create a basic one",
+    '[ -f app.json ] || echo "{}" > app.json',
```

Comment:
Is `{}` in `app.json` considered the bare minimum that works? Or is there a `app.json` template we recommend for folks with a bare React Native looking to transition to expo?

Reply:
You don't need an `app.json` for anything in Expo CLI, if one is needed, it will be generated. This may not be the case in EAS CLI.

Reply:
The EAS cli setup commands wont work without an `app.json` with `expo` object. I fixed this with a copy pastable script that creates this for users. The default is an `app.json` with no `expo` object with a bare React Native app, so it's worth covering this case in the docs. 

There are too many places in our code where we don't consider this particular case (`eas-cli` especially). Better to put this short term fix in our migration docs first, then go back and add better support for transitioning bare workflow apps later in our code.

Reply:
```
eas update:configure
EAS project not configured.
✔ Would you like to automatically create an EAS project for @brents/quin? … yes
✔ Created @brents/quin: https://expo.dev/accounts/brents/projects/quin on Expo
✖ Linking local project to EAS project 0f084c7f-d209-491d-97ac-c8c5c4c2c122
No config exists
    Error: update:configure command failed.
```

we should just create app.json if it doesn't exist, imo.